crates/noirc_abi/src/lib.rs

```diff
@@ -28,12 +30,26 @@ pub const MAIN_RETURN_NAME: &str = "return";
 /// support.
 pub enum AbiType {
     Field,
-    Array { length: u128, typ: Box<AbiType> },
-    Integer { sign: Sign, width: u32 },
-    Struct { fields: BTreeMap<String, AbiType> },
+    Array {
+        length: u64,
+        #[serde(rename = "type")]
+        typ: Box<AbiType>,
+    },
+    Integer {
+        sign: Sign,
+        width: u32,
+    },
+    Struct {
+        #[serde(
+            serialize_with = "serialization::serialize_struct_fields",
+            deserialize_with = "serialization::deserialize_struct_fields"
+        )]
+        fields: BTreeMap<String, AbiType>,
```

Comment:
Why not using directly a vector here as this would remove the need for the custom serialiser?
I suppose it is to avoid modifying the other part of the code, however I expect changes to the Abi to be more painful in the future so if we want to change it I would prefer we do it now. I don't have a strong opinion thought on map vs vector so I would be glad to get some feedback.